### PR TITLE
feat(dataviz): Init dataviz package

### PR DESCRIPTION
**Pull request by gjou-tlnd**
**What is the problem this PR is trying to solve?**

TDP and TDS have almost the same chart panel and their style need to be updated. Moreover, new products are using recharts instead of d3 when possible

**What is the chosen solution to this problem?**

Share charts in a new lib @talend/dataviz

**Please check if the PR fulfills these requirements**

* [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

**Comment by trob-tlnd** [comment]
why this file is empty ? packages/dataviz/jest.setup.js

File: packages/dataviz/src/components/BarChart/useBarChartHook.test.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { getPrimaryBarValue, getSecondaryBarValue } from './useBarChart.hook';
+
+describe('useBarChart utils', () => {
+	describe('getPrimaryBarValue', () => {
+		it('Should return filtered value when available', () => {
+			expect(
+				getPrimaryBarValue({
+					key: '',
```
Comment (trob-tlnd) [inline packages/dataviz/src/components/BarChart/useBarChartHook.test.ts:8]: do a global variable no ?

File: packages/dataviz/src/components/BarChart/HorizontalBarChart/HorizontalBarChart.component.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+						interval={0}
+						mirror
+					/>
+				</BarChart>
+			</FixedBarSizeWrapper>
+		</ResponsiveContainer>
+	);
+}
+
+export default Object.assign(HorizontalBarChart);
```
Comment (jmfrancois) [inline packages/dataviz/src/components/BarChart/HorizontalBarChart/HorizontalBarChart.component.tsx:99]: Please remove this empty Object.assign

File: packages/dataviz/src/components/Tooltip/Tooltip.component.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+	return (
+		<ul className={styles['dataviz-tooltip']}>
+			{entries.map(({ key, value }) => (
+				<li key={key}>
+					{key}: <span className={styles['dataviz-tooltip__value']}>{value}</span>
+				</li>
+			))}
+		</ul>
+	);
+}
```
Comment (jmfrancois) [inline packages/dataviz/src/components/Tooltip/Tooltip.component.tsx:17]: Based on the code it looks like a list more than a Tooltip.

File: .surge/index.html [hunk 4]
```diff
@@ -123,8 +123,22 @@ <h4 class="list-group-item-heading">Faceted Search</h4>
...
+                        >
+                        <a href="/@talend/dataviz/dist/TalendReactDataviz.min.js.report.html"
+                        >minified UMD</a
+                        >
+                    </div>
 				</div>
 			</div>
 		</div>
 	</body>
+</body>
```
Comment (frassinier) [inline .surge/index.html:143]:
```suggestion

```

File: packages/dataviz/.gitignore [hunk 5]
```diff
@@ -0,0 +1,2 @@
+i18n-extract
+*.log
```
Comment (frassinier) [inline packages/dataviz/.gitignore:2]: we can exclude src isn't it?

File: packages/dataviz/.storybook/preview.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { I18nextProvider } from 'react-i18next';
+import IconsProvider from "@talend/react-components/lib/IconsProvider";
```
Comment (frassinier) [inline packages/dataviz/.storybook/preview.js:3]: Do you use icons?

File: packages/dataviz/src/components/Tooltip/__snapshots__/Tooltip.component.test.tsx.snap [hunk 7]
```diff
@@ -0,0 +1,12 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`Tooltip Should render 1`] = `
+<ul class="theme-dataviz-tooltip">
```
Comment (frassinier) [inline packages/dataviz/src/components/Tooltip/__snapshots__/Tooltip.component.test.tsx.snap:4]: Please consider using a definition list for key-value pattern

File: packages/dataviz/package.json [hunk 8]
```diff
@@ -0,0 +1,76 @@
+{
+  "name": "@talend/dataviz",
```
Comment (jmfrancois) [inline packages/dataviz/package.json:2]: please name @talend/react-dataviz

File: packages/dataviz/package.json [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+  "version": "0.0.1",
+  "description": "Talend charts and visualization components",
+  "main": "lib/index.js",
+  "mainSrc": "src/index.ts",
+  "license": "Apache-2.0",
+  "scripts": {
+    "build": "talend-scripts build:ts:lib",
```
Comment (jmfrancois) [inline packages/dataviz/package.json:9]: please add the build:lib:umd also in prepublishOnly

**Review by frassinier — APPROVED** [review]
It seems to be a very valuable work even if it's hard to check everything, it's a definitively first step 💯 
I'm wondering if we can push it as a standalone repository, like the faceted-search, wdyt @Talend/frontend-admins ?

File: packages/dataviz/package.json [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+  },
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/Talend/ui.git"
+  },
+  "dependencies": {
+    "@talend/react-components": "^6.2.2",
+    "classnames": "^2.2.6",
+    "d3": "^6.3.0",
+    "d3-format": "^2.0.0",
```
Comment (jmfrancois) [inline packages/dataviz/package.json:37]: i didn t have found any usage of d3-format

File: packages/dataviz/src/components/Tooltip/Tooltip.component.tsx (hunk 3 above)
Comment (gjou-tlnd) [reply packages/dataviz/src/components/Tooltip/Tooltip.component.tsx:17]: indeed, it's a tooltipcontent to share styles between D3 and recharts based charts.
I'm renaming the component

File: packages/dataviz/.storybook/preview.js (hunk 6 above)
Comment (gjou-tlnd) [reply packages/dataviz/.storybook/preview.js:3]: right now? no.. but I forgot to replace blank spaces with the svg.. on it

File: .github/workflows/dataviz-autopublish.yml [hunk 11]
```diff
@@ -0,0 +1,60 @@
+name: Dataviz module auto publish
```
Comment (romainseb) [inline .github/workflows/dataviz-autopublish.yml:1]: Do we want a autopublish for this package or should it be part of the TUI lifecycle ?

File: packages/dataviz/LICENSE [hunk 12]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (romainseb) [inline packages/dataviz/LICENSE:1]: Are we in Apache licence on TUI ? 🤔

File: packages/dataviz/package.json [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+  "homepage": "https://github.com/Talend/ui/tree/master/packages/dataviz#readme",
+  "bugs": {
+    "url": "https://github.com/Talend/ui/issues"
+  },
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/Talend/ui.git"
+  },
+  "dependencies": {
+    "@talend/react-components": "^6.2.2",
```
Comment (romainseb) [inline packages/dataviz/package.json:35]: Should this be a peer dep ?

File: packages/dataviz/package.json [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+    "i18next-scanner": "^3.0.0",
+    "i18next-scanner-typescript": "^1.0.6",
+    "react": "^16.8.6",
+    "react-dom": "^16.8.6",
+    "react-i18next": "^10.10.0"
+  },
+  "peerDependencies": {
+    "@talend/bootstrap-theme": "^6.1.7",
+    "@talend/icons": "^6.2.2",
+    "i18next": "^19.8.4",
```
Comment (romainseb) [inline packages/dataviz/package.json:69]: We don't have this version of i18next in TUI for now, I think we should use the same version as we use 15.X for now in projects

File: packages/dataviz/package.json [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+    "react": "^16.14.0",
+    "react-i18next": "^10.10.0"
```
Comment (romainseb) [inline packages/dataviz/package.json:71]: Kind the same, not sure we got projects running this version of react-i18next (should be synchronised with the TUI ecosystem)

**Review by romainseb — COMMENTED** [review]
jest.setup.js file is empty

File: packages/dataviz/LICENSE (hunk 12 above)
Comment (gjou-tlnd) [reply packages/dataviz/LICENSE:1]: used the same license as faceted search and components

File: packages/dataviz/package.json (hunk 14 above)
Comment (gjou-tlnd) [reply packages/dataviz/package.json:69]: fixed

File: packages/dataviz/package.json (hunk 15 above)
Comment (gjou-tlnd) [reply packages/dataviz/package.json:71]: react-components is already requireing     "react-i18next": "^10.11.4"

**Review by jmfrancois — APPROVED** [review]

**Pull request merged**